### PR TITLE
feat: efs deployment type

### PR DESCRIPTION
**Pull request by seanr15**
## Description
Add CloudFormation templates that deploy and configure EFS/FSx with a Determined Cluster on AWS

## Test Plan
Manually tested on AWS

### CODE REVIEW

File: deploy/determined_deploy/aws/deployment_types/efs.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+from typing import List
+
+import boto3
+import pkg_resources
+
+from determined_deploy.aws import aws, constants
+from determined_deploy.aws.deployment_types import base
+
+
+class EFS(base.DeterminedDeployment):
```
Comment (aaron276h) [inline deploy/determined_deploy/aws/deployment_types/efs.py:10]: It seems like this and the `FSx` class are identical, could we avoid duplicating this code?

**Comment by davidhershey** [comment]
Is there a world in which efs / fsx could just be arguments conditioned on the VPC deployment type?  And if you try to use those flags without VPC it throws an error?

**Comment by seanr15** [comment]
> Is there a world in which efs / fsx could just be arguments conditioned on the VPC deployment type? And if you try to use those flags without VPC it throws an error?

That was the original plan, but it would require adding conditional statements in the template and build different resources based on that. It's definitely more complicated, and I would rather tackle that down the road once it sees more usage.

**Comment by davidhershey** [comment]
> > Is there a world in which efs / fsx could just be arguments conditioned on the VPC deployment type? And if you try to use those flags without VPC it throws an error?
> 
> That was the original plan, but it would require adding conditional statements in the template and build different resources based on that. It's definitely more complicated, and I would rather tackle that down the road once it sees more usage.

Works for me.

File: docs/how-to/installation/aws.txt [hunk 2]
```diff
@@ -96,6 +96,14 @@ using the ``--deployment-type`` argument (e.g., ``det-deploy aws
 -  ``vpc``: The VPC deployment creates a separate VPC with public
    subnets; the Determined cluster is deployed into these subnets.
 
+-  ``efs``: The EFS deployment creates an EFS drive and a Determined
```
Comment (aaron276h) [inline docs/how-to/installation/aws.txt:99]: non-blocking: link to [efs link](https://aws.amazon.com/efs/)

File: docs/how-to/installation/aws.txt [hunk 3]
```diff
@@ -96,6 +96,14 @@ using the ``--deployment-type`` argument (e.g., ``det-deploy aws
...
+   cluster into a separate VPC. The EFS drive attaches to agent
+   instances at `/mnt/efs`.
+
+-  ``fsx``: The FSX deployment creates a Lustre FSx drive and a
```
Comment (aaron276h) [inline docs/how-to/installation/aws.txt:103]: same

**Review by aaron276h — APPROVED** [review]
Great work 🍾 Can you also add release notes.

**Pull request merged**